neonvm/controllers/virtualmachinemigration_controller.go

```diff
@@ -218,8 +218,31 @@ func (r *VirtualMachineMigrationReconciler) Reconcile(ctx context.Context, req c
 		targetRunner := &corev1.Pod{}
 		err := r.Get(ctx, types.NamespacedName{Name: migration.Status.TargetPodName, Namespace: vm.Namespace}, targetRunner)
 		if err != nil && apierrors.IsNotFound(err) {
+			enableSSH := false
+			if vm.Spec.EnableSSH != nil && *vm.Spec.EnableSSH {
+				enableSSH = true
+			}
+			var sshSecret *corev1.Secret
+			if enableSSH {
+				// We require the SSH secret to exist because we cannot unmount and
+				// mount the new secret into the VM after the live migration. If a
+				// VM's SSH secret is deleted accidentally then live migration is
+				// not possible.
+				if len(vm.Status.SSHSecretName) == 0 {
+					err := errors.New("VM .Status.SSHSecretName is empty")
+					log.Error(err, "Failed to get VM's SSH Secret")
+					return ctrl.Result{}, err
+				}
```

Comment:
Not sure if it's possible to get here, but I don't really see any issue with checking this

Reply:
As for now yes, but things might change in the future so it's better to check it.